pyproject.toml

```diff
@@ -26,24 +26,24 @@ classifiers=[
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-"numpy",
-"accelerate>=0.20.3",
-"transformers",
-"torch",
-"sentencepiece",
-"tokenizers>=0.13.3",
-"tqdm",
+"numpy>=1.26.4,<2.0",
```

Comment:
Note that numpy is now at 2.0 as of a few days ago. I have found that numpy 2.0 works still with tuning and the stack of deps. 

Although I did run into an issue running unit tests locally if torch==2.2.2 and numpy==2.0.0 then I got error `RuntimeError: Numpy is not available` from the unit tests. See [slack thread](https://ibm-research.slack.com/archives/C06MZPQ4ENL/p1718827390959019?thread_ts=1718820929.017379&cid=C06MZPQ4ENL). But as you can see in unittests it succeeds
- [ ran today](https://github.com/foundation-model-stack/fms-hf-tuning/actions/runs/9583556592/job/26425112934) passed with numpy 2.0.0, transformers 4.41.2, torch 2.3.1
- [older transformers](https://github.com/foundation-model-stack/fms-hf-tuning/actions/runs/9587647439/job/26438105396?pr=204) numpy 2.0.0, transformers 4.40.2, torch 2.3.1

Reply:
In addition, the pytorch community did test numpy 2.0.0rc1 from this [issue](https://github.com/pytorch/pytorch/issues/107302) and deemed it suitable and they have updated their deps to use numpy 2.0. But the upgrade should be evaluated further as it does contain breaking changes, see [release notes](https://numpy.org/devdocs/release/2.0.0-notes.html).

Reply:
The latest version of aim has a dependency to [`numpy<2,>=1.12.0`](https://github.com/aimhubio/aim/blob/fddfe1ed48908d5a5db60ca27ec675c039fcdf41/setup.py#L60). So I think it's best to keep `numpy>=1.26.4,<2.0` for now.

Reply:
good to know, thanks Vassilis!